tests/should_succeed/and_or_not.jou

```diff
@@ -0,0 +1,95 @@
+declare putchar(ch: int) -> int
+declare printf(format: byte*, ...) -> int
+
+def side_effect(letter: byte, value: bool) -> bool:
+    putchar(letter)
+    return value
+
+def main() -> int:
+    # Output: 1000
+    printf("%d", 1+2==3 and 1+2==3)
+    printf("%d", 1+2==3 and 1+2==4)
+    printf("%d", 1+2==4 and 1+2==3)
+    printf("%d", 1+2==4 and 1+2==4)
+    putchar('\n')
+
+    # Output: 1110
+    printf("%d", 1+2==3 or 1+2==3)
+    printf("%d", 1+2==3 or 1+2==4)
+    printf("%d", 1+2==4 or 1+2==3)
+    printf("%d", 1+2==4 or 1+2==4)
+    putchar('\n')
+
+    # Output: 01
+    printf("%d", not 1+2==3)
+    printf("%d", not 1+2==4)
+    putchar('\n')
+
+    # Anything with "False and ..." contains code that never runs: the "..." part
+    # is not evaluated at all because the result will be False anyway.
```

Comment:
```suggestion
    # is not evaluated at all because the result will be False anyway.
    #
    # Technically "True and ..." also contains code that never runs: the left side
    # of "and" is never False, but the compiler still generates code to handle that
    # case (the generated code sets the result of the "and" operation to False).
    # Showing a warning about the generated code would probably be more
    # confusing than helpful, so the compiler hides it.
    #
    # Similar things apply to "or".
```